src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double DEFAULT_LOAD = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] innerArray;
+    private int size;
+
+    public MyHashMap() {
+        innerArray = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        grow();
+        int index = getIndex(key);
+        Node<K, V> tempNode = innerArray[index];
+        if (tempNode == null) {
+            innerArray[index] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        Node<K, V> prevNode = tempNode;
+        while (tempNode != null) {
+            if (Objects.equals(tempNode.key, key)) {
+                tempNode.value = value;
+                return;
+            }
+            prevNode = tempNode;
+            tempNode = tempNode.next;
+        }
+        prevNode.next = new Node<>(key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> tempNode = innerArray[getIndex(key)];
+        if (tempNode == null) {
+            return null;
+        }
+        if (Objects.equals(tempNode.key, key)) {
+            return tempNode.value;
+        }
+        while (tempNode != null) {
+            if (Objects.equals(tempNode.key, key)) {
+                return tempNode.value;
+            }
+            tempNode = tempNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        private Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    private int getIndex(K key) {
+        return getHash(key) % innerArray.length;
+    }
+
+    private int getHash(K key) {
+        if (key == null) {
+            return 0;
+        }
+        int hash = key.hashCode();
+        return hash < 0 ? -hash : hash;
+    }
+
+    private void grow() {
+        int length = innerArray.length;
+        if (size >= length * DEFAULT_LOAD) {
```

Comment:
I think better create private variable **currentLoadFactor =  innerArray.length**  which you will calculate only when you resize you innerArray, and do if (size >= currentLoadFactor) . Because now you do redundant calculate and redundant call **innerArray.length**  every time, when you  put new element to you HashMap

Reply:
Well yes, but it's debatable. Length is simple request that executes only once per call and reduces amount of fields.
I see you've approved, so let us see mentor's opinion)